msal/src/cache/java/com/microsoft/identity/client/BaseTokenCacheItem.java

```diff
@@ -31,13 +31,13 @@
 abstract class BaseTokenCacheItem {
 
     @SerializedName("client_id")
-    final String mClientId;
+    String mClientId;
```

Comment:
@iambmelt @weijjia This PR solves the unsafe allocator issue: https://github.com/google/gson/blob/09cbc8073bb2bc60aeed6b831be4473daaf915ae/gson/src/main/java/com/google/gson/internal/UnsafeAllocator.java#L42.

To prevent introducing a new one, I removed the finals: https://github.com/google/gson/blob/9e44d60b83d667d96150eeff4e3f04cd4f157e9a/gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java#L157.

The `ReflectiveTypeAdapterFactory` is used in the `Gson` class here: https://github.com/google/gson/blob/92b52d25d844b81620a04917e5fa164d44b9c8a5/gson/src/main/java/com/google/gson/Gson.java#L252.

Basically I would like this to use as little of reflection as we can.